### PR TITLE
feat(crepe): derive toolbar shortcuts from keymaps

### PR DESCRIPTION
**Pull request by Saul-Mirone**
- [x] I read the contributing guide
- [x] I agree to follow the code of conduct

## Summary

Follow-up to #2435 (toolbar accessible names). That PR gave `ToolbarItem` the `shortcut` / `ariaKeyshortcuts` fields but left the built-in items advertising nothing, because a shortcut is bound in the preset keymaps and spelling it again per UI region is duplicative and drifts from the real binding.

This makes the **keymap the single source of truth**. A `ToolbarItem` can now carry a `keymap` reference to the entry that binds its command; the tooltip `shortcut` and `aria-keyshortcuts` are derived from it — per platform, and rebind-aware, since the keymap slice is user-customizable. The four built-in formatting buttons ship these references:

| item | keymap | shortcut |
|---|---|---|
| bold | `strongKeymap` / `ToggleBold` | `Mod-b` |
| italic | `emphasisKeymap` / `ToggleEmphasis` | `Mod-i` |
| code | `inlineCodeKeymap` / `ToggleInlineCode` | `Mod-e` |
| strikethrough | `strikethroughKeymap` / `ToggleStrikethrough` | `Mod-Alt-x` |

Design notes:

- New `packages/crepe/src/utils/keyboard-shortcut.ts`: `keymapRef()` (a generic factory that keeps the entry name compile-checked and localizes the one cast), a pure `formatKeymapShortcut(shortcut, mac)` that mirrors prosemirror-keymap's `normalizeKeyName` (`Mod` → `Meta` on macOS / `Control` elsewhere), and `resolveKeymapShortcut()`.
- Platform detection reuses prosemirror's `browser.mac` (`@milkdown/kit/prose`) — no hand-rolled `navigator` check, and the AI feature's `detectModSymbol` is intentionally left alone (different semantics / SSR default).
- Resolution happens in `getGroups` as a `ctx`-guarded pass that fills the fields with `??=`, so an explicitly configured string still wins — the escape hatch for shortcuts with no milkdown keymap. `component.tsx` is unchanged.
- `keymapRef` / `KeymapRef` / `ToolbarItem` are exported from `@milkdown/crepe/feature/toolbar` so a custom item can point at a keymap instead of re-spelling the shortcut.
- Scope is the floating Toolbar. The util is region-agnostic; the TopBar can adopt it once it gains accessible names (separate change).

## How did you test this change?

- New `keyboard-shortcut.spec.ts`: table tests for both platforms, input-order normalization, the `-` / `Space` / `ArrowUp` edge cases.
- Extended `toolbar.spec.ts` against a real `Crepe` editor ctx: built-ins derive from the keymap; a `ctx.set(strongKeymap.key, …)` rebind flows through; an explicit string overrides the derived value; a custom item reuses a keymap ref; items with no keymap advertise nothing.
- `tsc --noEmit` clean, `oxlint --deny-warnings` clean, full crepe suite green (80 tests).

### CODE REVIEW

File: packages/crepe/src/utils/keyboard-shortcut.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+/// Returns `undefined` when the entry carries no shortcut. Only the first
+/// binding is used when an entry lists several.
+export function resolveKeymapShortcut(
+  ctx: Ctx,
+  ref: KeymapRef
+): ResolvedShortcut | undefined {
+  const entry = ctx.get(ref.slice)[ref.entry]
+  const shortcut = entry ? [entry.shortcuts].flat()[0] : undefined
+  if (!shortcut) return undefined
+  return formatKeymapShortcut(shortcut, browser.mac)
```
[the remark this answers is not in the document]
Comment (Saul-Mirone) [reply packages/crepe/src/utils/keyboard-shortcut.ts:153]: Good catch — fixed in 773c7201. `resolveKeymapShortcut` now returns `undefined` when `ctx.isInjected(ref.slice)` is false, so a custom item pointing at an unloaded keymap no longer crashes toolbar construction. Added a regression test.

File: docs/api/crepe.md [hunk 2]
```diff
@@ -413,20 +413,45 @@ type ToolbarItem = {
...
+import { strongKeymap } from '@milkdown/kit/preset/commonmark'
+
+// A second entry for an existing command reuses the one binding — no need to
+// re-spell ⌘B here or anywhere else it appears.
+builder.addGroup('custom', 'Custom').addItem('bold', {
+  icon: boldIcon,
+  label: 'Bold',
+  keymap: keymapRef(strongKeymap.key, 'ToggleBold'),
+  active: (ctx) => …,
+  onRun: (ctx) => …,
```
[the remark this answers is not in the document]
Comment (Saul-Mirone) [reply docs/api/crepe.md]: Fixed in 773c7201 — the `typescript` snippets now use valid expressions instead of the `…` placeholder.
